src/PlusImageProcessing/vtkPlusTransverseProcessEnhancer.cxx

```diff
@@ -337,49 +337,93 @@ PlusStatus vtkPlusTransverseProcessEnhancer::ReadConfiguration(vtkXMLDataElement
-  }
-
-  if (this->ErosionEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(erosionParameters, processingElement, "Erosion");
-    erosionParameters->SetVectorAttribute("ErosionKernelSize", 2, this->ErosionKernelSize);
-  }
-
-  if (this->DilationEnabled)
-  {
-    XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(dilationParameters, processingElement, "Dilation");
-    dilationParameters->SetVectorAttribute("DilationKernelSize", 2, this->DilationKernelSize);
-  }
-
-  return PLUS_SUCCESS;
+	XML_VERIFY_ELEMENT(processingElement, this->GetTagName());
+
+	XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(imageProcessingOperations, processingElement, "ImageProcessingOperations");
+
+	if (this->ConvertToLinesImage){
+		imageProcessingOperations->SetAttribute("ConvertToLinesImage", "True");
+	}
+	else{
+		imageProcessingOperations->SetAttribute("ConvertToLinesImage", "False");
+	}
+
+	if (this->GaussianEnabled)
+	{
+		imageProcessingOperations->SetAttribute("GaussianEnabled", "True");
+		XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(gaussianParameters, imageProcessingOperations, "GaussianSmoothing");
+		gaussianParameters->SetDoubleAttribute("GaussianStdDev", this->GaussianStdDev);
+		gaussianParameters->SetDoubleAttribute("GaussianKernelSize", this->GaussianKernelSize);
+	}
+	else{
+		imageProcessingOperations->SetAttribute("GaussianEnabled", "False");
+	}
+
+	if (this->ThresholdingEnabled)
+	{
+		imageProcessingOperations->SetAttribute("ThresholdingEnabled", "True");
+		XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(thresholdingParameters, imageProcessingOperations, "Thresholding");
+		thresholdingParameters->SetDoubleAttribute("ThresholdInValue", ThresholdInValue);
+		thresholdingParameters->SetDoubleAttribute("ThresholdOutValue", ThresholdOutValue);
+		thresholdingParameters->SetDoubleAttribute("LowerThreshold", LowerThreshold);
+		thresholdingParameters->SetDoubleAttribute("UpperThreshold", UpperThreshold);
+	}
+	else{
+		imageProcessingOperations->SetAttribute("ThresholdingEnabled", "False");
+	}
+
+	if (this->EdgeDetectorEnabled)
+	{
+		imageProcessingOperations->SetAttribute("EdgeDetectorEnabled", "True");
+	}
+	else{
+		imageProcessingOperations->SetAttribute("EdgeDetectorEnabled", "False");
+	}
+
+	if (this->IslandRemovalEnabled){
+		imageProcessingOperations->SetAttribute("IslandRemovalEnabled", "True");
+		XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(islandRemovalParameters, imageProcessingOperations, "IslandRemoval");
+		islandRemovalParameters->SetIntAttribute("IslandAreaThreshold", IslandAreaThreshold);
+	}
+	else{
+		imageProcessingOperations->SetAttribute("IslandRemovalEnabled", "False");
+	}
+
+	if (this->ErosionEnabled)
+	{
+		imageProcessingOperations->SetAttribute("ErosionEnabled", "True");
+		XML_FIND_NESTED_ELEMENT_CREATE_IF_MISSING(erosionParameters, imageProcessingOperations, "Erosion");
+		erosionParameters->SetVectorAttribute("ErosionKernelSize", 2, this->ErosionKernelSize);
+	}
+	else{

```

Comment:
Follow coding style of Plus:
* 2 spaces for indentation
* open/closing braces are in separate lines